When wrapping with Emmet, consider the document language when setting the syntax.

mattkwiecien opened the pull request:
`doWrapping` was previously setting `syntax = 'html` when wrapping with abbreviations.  This was resulting in bad class attribute names for `jsx` files.

* Added a call to an existing function that determines the syntax based on the `args` parameter passed to `doWrapping`.  
* It's possible for this `args` parameter to be undefined, so added a guard clause to ensure that it is 
  * not undefined, and 
  * the language property is set by the current `editor.document.languageId`.  

Now the `syntax` value is based on the return of `getSyntaxFromArgs`, and if the result of this function is undefined, default to `'html'`.

This fixes https://github.com/Microsoft/vscode/issues/67098.

ramya-rao-a commented:
@mattkwiecien Nice! Do you mind adding a test case for this in `wrapWithAbbreviation.test.ts` file

mattkwiecien commented:
Absolutely!  Will do either tonight or tomorrow morning.

msftclas commented:
[![CLA assistant check](https://cla.opensource.microsoft.com/pull/badge/not_signed)](https://cla.opensource.microsoft.com/Microsoft/vscode?pullRequest=68326) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.opensource.microsoft.com/Microsoft/vscode?pullRequest=68326) before we can accept your contribution.<br/><br/>:x: mattkwiecien [sign now](https://cla.opensource.microsoft.com/Microsoft/vscode?pullRequest=68326)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.opensource.microsoft.com/check/Microsoft/vscode?pullRequest=68326) it.</sub>

mattkwiecien commented:
Added two unit tests for `wrapWithAbbreviation` and `wrapIndividualLinesWithAbbreviation` in `wrapWithAbbreviation.test.ts` for when the file extension is 'jsx'.

While I was in there, I noticed that a lot of the unit tests were re-writing existing logic in the `testWrapWithAbbreviation` 
 function (and the newly added `testWrapIndividualLinesWithAbbreviation` function), so I refactored those tests to call that function instead of copying the same logic within each unit test.

mattkwiecien commented:
@octref / @ramya-rao-a 
Am I good to merge this PR?

octref commented on extensions/emmet/src/test/wrapWithAbbreviation.test.ts:
```diff
@@ -63,6 +63,22 @@ const wrapInlineElementExpectedFormatFalse = `
+		<div className="hello2"><li class="item2">$hithere</li></div>
+	</ul>
+`;
+
```
Can you put these together with their tests?

mattkwiecien replied:
Absolutely!

octref commented:
Just a small nitpick, looks good to me. Thanks!

octref commented:
Awesome contribution, thank you!

mattkwiecien commented:
Thanks, really looking forward to contributing more to this great project :)
